mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/task/RecomputeCurrentQuotasService.java

```diff
@@ -0,0 +1,87 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.store.mail.task;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.model.CurrentQuotas;
+import org.apache.james.mailbox.model.QuotaOperation;
+import org.apache.james.mailbox.model.QuotaRoot;
+import org.apache.james.mailbox.quota.UserQuotaRootResolver;
+import org.apache.james.mailbox.store.quota.CurrentQuotaCalculator;
+import org.apache.james.mailbox.store.quota.StoreCurrentQuotaManager;
+import org.apache.james.task.Task;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.util.streams.Iterators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RecomputeCurrentQuotasService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecomputeCurrentQuotasService.class);
+
+    private final UsersRepository usersRepository;
+    private final StoreCurrentQuotaManager storeCurrentQuotaManager;
+    private final CurrentQuotaCalculator currentQuotaCalculator;
+    private final UserQuotaRootResolver userQuotaRootResolver;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    RecomputeCurrentQuotasService(UsersRepository usersRepository,
+                                  StoreCurrentQuotaManager storeCurrentQuotaManager,
+                                  CurrentQuotaCalculator currentQuotaCalculator,
+                                  UserQuotaRootResolver userQuotaRootResolver,
+                                  SessionProvider sessionProvider) {
+        this.usersRepository = usersRepository;
+        this.storeCurrentQuotaManager = storeCurrentQuotaManager;
+        this.currentQuotaCalculator = currentQuotaCalculator;
+        this.userQuotaRootResolver = userQuotaRootResolver;
+        this.sessionProvider = sessionProvider;
+    }
+
+    Task.Result recomputeCurrentQuotas() throws UsersRepositoryException {
+        return Iterators.toStream(usersRepository.list())
```

Comment:
Is `/mailboxes` really the right endpoint for this task, as I start from users here? Or is there an other way? And how about the `/quotas/users` endpoint?

Reply:
Maybe `/users?action=RecomputeQuotas` but good remark, yes ^^